cmd/bosun/expr/expr.go

```diff
@@ -212,6 +212,20 @@ func (r ResultSlice) DescByValue() ResultSlice {
 	return c
 }
 
+// Filter strips out results if their tagset doesn't conform to the given key/value pair restrictions
+// It updates the input slice in place.
+func (r ResultSlice) Filter(filter opentsdb.TagSet) ResultSlice {
+	lastGoodIndex := -1
+
+	for _, res := range r {
+		if res.Group.Compatible(filter) {
+			lastGoodIndex++
+			r[lastGoodIndex] = res
```

Comment:
Assume `a` is an existing ResultSlice:

```
b := a
a = a.Filter(opentsdb.TagSet{"b": "c"})
```

Now b is messed up and longer than a and contains duplicates and is missing things it used to have. The sort case is different because it's just changing the order of the things, but keeping the same things.